app/views/incomes/_form.html.erb

```diff
@@ -0,0 +1,47 @@
+<%= form_for @income, url: url_for(controller: 'incomes', action: action) do |form| %>
+  <% if @income.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@income.errors.count, "error") %> prohibited this transaction from being saved:</h2>
+
+      <ul>
+        <% @income.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```

Comment:
Create a shared partial and move this part of the view.
Path: `app/view/shared/_form_errors.html.erb`

Call like this:
```
<%= render 'shared/form_errors', object: @income %>
```

Inside the partial:
```
<% if object.errors.any? %>
    <div id="error_explanation">
    ...
```

Reply:
Then refactor other forms as well.